model/cells/sir.hpp

```diff
@@ -0,0 +1,107 @@
+//
+// Created by binybrion on 6/30/20.
+//
+
+#ifndef PANDEMIC_HOYA_2002_SIR_HPP
+#define PANDEMIC_HOYA_2002_SIR_HPP
+
+#include <iostream>
+#include <nlohmann/json.hpp>
+
+struct sir {
+    unsigned int population_density;
+    unsigned int phase;
+    int num_inf;
+    int num_rec;
+    float susceptible;
+    float initial_infected;
+    std::vector<float> infected;
+    std::vector<float> recovered;
+
+    float border_length;
+    float land_area;
+
+    sir(unsigned int popDensity, unsigned int phase, std::vector<float> infected, std::vector<float> recovered, float border_length, float land_area)
+            :
+                population_density{popDensity},
+                phase{phase},
+                infected{std::move(infected)},
+                recovered{std::move(recovered)},
+                border_length{border_length},
+                land_area{land_area} {
+        initialize();
+    }
+
+    sir()
+            :
+                population_density{0},
+                phase{0},
+                susceptible{1} {
+
+    }
+
+    void initialize() {
+
+        initial_infected = infected.front();
+
+        num_inf = infected.size();
+
+        num_rec = recovered.size();
+
+        susceptible = 1 - initial_infected;
+    }
+
+    bool operator!=(const sir &other) {
+        bool neq = population_density != other.population_density ||
+                   phase != other.phase ||
+                   susceptible != other.susceptible;
+
+        int i = 0;
+
+        while (!neq && i < num_inf) {
+            neq = infected[i] != other.infected[i];
```

Comment:
Be careful here, you are overwriting the `neq` variable. 

I recommend you to return `true` as soon as you find a field that does not match with the other field, so we could speed up a little bit the simulation.

Reply:
You could also change the `neq = ...` to `neq |= ...`